### PR TITLE
Add custom domain and route

### PR DESCRIPTION
**Pull request by merrilmathew**
### Context
Terraform support for Azure front door custom domain is now available. This commit adds the missing FD reources as part of custom domain deployment.

### CODE REVIEW

File: domains/environment_domains/front_door.tf [hunk 1]
```diff
@@ -22,12 +22,90 @@ resource "azurerm_cdn_frontdoor_origin_group" "main" {
...
+  dns_zone_id              = data.azurerm_dns_zone.main.id
+  host_name                = each.key == "apex" ? "${var.zone}" : "${each.key}.${var.zone}"
+  tls {
+    certificate_type    = "ManagedCertificate"
+    minimum_tls_version = "TLS12"
+  }
+}
+
+resource "azurerm_dns_txt_record" "main" {
+  for_each            = { for k, v in toset(var.domains) : k => v if v != "apex" }
```
Comment (NickGraham101) [inline domains/environment_domains/front_door.tf:46]: Do we need to use the key, value syntax here?  I think we might be able to filter using the simpler syntax described [here](https://developer.hashicorp.com/terraform/language/expressions/for#filtering-elements)

File: domains/environment_domains/front_door.tf [hunk 2]
```diff
@@ -22,12 +22,90 @@ resource "azurerm_cdn_frontdoor_origin_group" "main" {
...
+  name                = join(".", ["_dnsauth", "${each.key}"])
```
Comment (NickGraham101) [inline domains/environment_domains/front_door.tf:47]: Won't the `key` be the index of each item in the set, ie an int when we want the string value that corresponds to the subdomain as described [here](https://learn.microsoft.com/en-us/azure/frontdoor/front-door-how-to-onboard-apex-domain?pivots=front-door-standard-premium#onboard-the-custom-domain-to-your-front-door)?

File: domains/environment_domains/front_door.tf [hunk 3]
```diff
@@ -22,12 +22,90 @@ resource "azurerm_cdn_frontdoor_origin_group" "main" {
...
+  resource_group_name = var.resource_group_name
+  ttl                 = 3600
+
+  record {
+    value = azurerm_cdn_frontdoor_custom_domain.main[each.key].validation_token
+  }
+}
+
+resource "azurerm_dns_txt_record" "apex" {
+  for_each            = { for k, v in toset(var.domains) : k => v if k == "apex" }
```
Comment (NickGraham101) [inline domains/environment_domains/front_door.tf:58]: Would also need changing if my [comment above](https://github.com/DFE-Digital/terraform-modules/pull/3/files#r1007068467) is valid

File: domains/environment_domains/front_door.tf (hunk 2 above)
Comment (merrilmathew) [reply domains/environment_domains/front_door.tf:47]: key appears to be the string value. i.e apex and www in ["apex", "www"].

**Review by NickGraham101 — APPROVED** [review]

**Pull request merged**